src/main/groovy/com/blackducksoftware/integration/hub/detect/codelocation/CodeLocationNameService.java

```diff
@@ -49,13 +49,14 @@
     @Autowired
     private CodeLocationNameProvider3 codeLocationNameProvider3;
 
-    public CodeLocationName createBomToolName(final String sourcePath, final String projectName, final String projectVersionName, final BomToolType bomToolType, final String prefix, final String suffix) {
-        final CodeLocationName codeLocationName = new CodeLocationName(projectName, projectVersionName, bomToolType, sourcePath, null, prefix, suffix, CodeLocationType.BOM);
+    public CodeLocationName createBomToolName(final String sourcePath, final String projectName, final String projectVersionName, final BomToolType bomToolType, final String prefix, final String suffix,
+            final List<String> additionalNamePieces) {
+        final CodeLocationName codeLocationName = new CodeLocationName(projectName, projectVersionName, bomToolType, sourcePath, null, prefix, suffix, additionalNamePieces, CodeLocationType.BOM);
         return codeLocationName;
     }
 
     public CodeLocationName createScanName(final String sourcePath, final String scanTargetPath, final String projectName, final String projectVersionName, final String prefix, final String suffix) {
-        final CodeLocationName codeLocationName = new CodeLocationName(projectName, projectVersionName, null, sourcePath, scanTargetPath, prefix, suffix, CodeLocationType.SCAN);
+        final CodeLocationName codeLocationName = new CodeLocationName(projectName, projectVersionName, null, sourcePath, scanTargetPath, prefix, suffix, null, CodeLocationType.SCAN);
         return codeLocationName;
```

Comment:
Im not sure if we want to update generateBomToolCurrent(final CodeLocationName codeLocationName) and generateScanCurrent(final CodeLocationName codeLocationName) or not